File: programming files/sum_even_fibonacci.java
public class sum_even_fibonacci {
    public static void main(String[] args) {
         // initialization
        int a = 0, b = 1;  // first two fibonacci numbers
        int sumEven = 0;
        while (b < 4000000) {    // numbers to 4 million
            if (b % 2 == 0) {    // even of those
                sumEven += b;    // sum them
            }
            int temp = b;
            b = a + b;  // next fibonacci number = [-1] + [-2]
            a = temp;   // move other counter to Fib.[-1] from [-2]
        }
        System.out.println(sumEven);
    }
}

// answer should be 4613732
// answer is correct.

File: programming files/sum_even_fibonacci.py
a, b = 0, 1
sum_even = 0
# Only sum fibonacci values up to 4,000,000
while b < 4000000:
    if b % 2 == 0:
        sum_even += b
    a, b = b, a+b

print(sum_even)

# answer should be 4613732
